src/worker.js

```diff
@@ -8,13 +8,53 @@
 // This is the entry point file that is loaded first by each Web Worker
 // that executes pthreads on the Emscripten application.
```

Comment:
IIUC all the changes to this file (or most of them anyway) are part of making worker run in strict mode?   Can split this into its own PR?

Reply:
Most of them, yeah. I could split them out, but this PR would still depend on them, as otherwise output is invalid.

Reply:
Aside from code having to be carefully updated - no.

Reply:
Extracted here: https://github.com/emscripten-core/emscripten/pull/14137